alacritty_terminal/src/ansi.rs

```diff
@@ -1446,140 +1446,184 @@ mod tests {
+            parser.advance(&mut handler, *byte, &mut io::sink());
         }
-    }
 
-    struct CharsetHandler {
-        index: CharsetIndex,
-        charset: StandardCharset,
-    }
+        assert!(handler.identity_reported);
+        handler.reset_state();
+
+        let bytes: &[u8] = &[0x1b, b'#', b'Z'];
+
+        let mut parser = Processor::new();
+        let mut handler = MockHandler::default();
 
-    impl Default for CharsetHandler {
-        fn default() -> CharsetHandler {
-            CharsetHandler { index: CharsetIndex::G0, charset: StandardCharset::Ascii }
+        for byte in &bytes[..] {
+            parser.advance(&mut handler, *byte, &mut io::sink());
         }
+
+        assert!(!handler.identity_reported);
+        handler.reset_state();
     }
 
-    impl Handler for CharsetHandler {
-        fn configure_charset(&mut self, index: CharsetIndex, charset: StandardCharset) {
-            self.index = index;
-            self.charset = charset;
-        }
+    #[test]
+    fn parse_truecolor_attr() {
+        static BYTES: &[u8] = &[
+            0x1b, b'[', b'3', b'8', b';', b'2', b';', b'1', b'2', b'8', b';', b'6', b'6', b';',
+            b'2', b'5', b'5', b'm',
+        ];
 
-        fn set_active_charset(&mut self, index: CharsetIndex) {
-            self.index = index;
+        let mut parser = Processor::new();
+        let mut handler = MockHandler::default();
+
+        for byte in &BYTES[..] {
+            parser.advance(&mut handler, *byte, &mut io::sink());
         }
+
+        let spec = Rgb { r: 128, g: 66, b: 255 };
+
+        assert_eq!(handler.attr, Some(Attr::Foreground(Color::Spec(spec))));
     }
 
-    impl TermInfo for CharsetHandler {
-        fn lines(&self) -> Line {
-            Line(200)
-        }
+    /// No exactly a test; useful for debugging
+    #[test]
+    fn parse_zsh_startup() {
+        static BYTES: &[u8] = &[
+            0x1b, b'[', b'1', b'm', 0x1b, b'[', b'7', b'm', b'%', 0x1b, b'[', b'2', b'7', b'm',
+            0x1b, b'[', b'1', b'm', 0x1b, b'[', b'0', b'm', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ', b' ',
+            b' ', b' ', b' ', b'\r', b' ', b'\r', b'\r', 0x1b, b'[', b'0', b'm', 0x1b, b'[', b'2',
+            b'7', b'm', 0x1b, b'[', b'2', b'4', b'm', 0x1b, b'[', b'J', b'j', b'w', b'i', b'l',
+            b'm', b'@', b'j', b'w', b'i', b'l', b'm', b'-', b'd', b'e', b's', b'k', b' ', 0x1b,
+            b'[', b'0', b'1', b';', b'3', b'2', b'm', 0xe2, 0x9e, 0x9c, b' ', 0x1b, b'[', b'0',
```

Comment:
Some bytes were left because they made no sense in char from (I was converting with python's chr func).

Reply:
Yeah, that's why I mentioned printable characters specifically. So that's perfectly fine.